specification/monitor/resource-manager/microsoft.insights/stable/2018-01-01/metricDefinitions_API.json

```diff
@@ -0,0 +1,430 @@
+        },
+        "namespace": {
+          "type": "string",
+          "description": "the namespace the metric blongs to."
+        },
+        "name": {
+          "$ref": "#/definitions/LocalizableString",
+          "description": "the name and the display name of the metric, i.e. it is a localizable string."
+        },
+        "unit": {
+          "$ref": "#/definitions/Unit",
+          "description": "the unit of the metric."
+        },
+        "primaryAggregationType": {
+          "$ref": "#/definitions/AggregationType",
+          "description": "the primary aggregation type value defining how to use the values for display.",
+        },
+		"supportedAggregationTypes": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AggregationType"
+          },
+          "description": "the collection of what aggregation types are supported."
+        },
+        "metricAvailabilities": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MetricAvailability"
+          },
+          "description": "the collection of what aggregation intervals are available to be queried."
+        },
+        "id": {
+          "type": "string",
+          "description": "the resource identifier of the metric definition."
+        },
+        "dimensions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LocalizableString"
+          },
+          "description": "the name and the display name of the dimension, i.e. it is a localizable string."
+        }
+      },
+      "description": "Metric definition class specifies the metadata for a metric."
+    },
+    "MetricDefinitionCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MetricDefinition"
+          },
+          "description": "the values for the metric definitions."
+        }
+      },
+      "required": ["value"],
+      "description": "Represents collection of metric definitions."
+    },
+    "ErrorResponse": {
+      "description": "Describes the format of Error response.",
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "Error code",
+          "type": "string"
+        },
+        "message": {
+          "description": "Error message indicating why the operation failed.",
+          "type": "string"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "Client Api Version."
+    },
+    "ResourceUriParameter": {
+      "name": "resourceUri",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The identifier of the resource.",
+      "x-ms-parameter-location": "method",
+      "x-ms-skip-url-encoding": true
+    },
+	"MetricNamespaceParameter": {
```

Comment:
>MetricNamespaceParameter [](start = 2, length = 24)

Is it expected to generate this parameter as a method or a client parameter? (My guess is it is expected to be a query parameter)

Reply:
If it is expected to be part of the method, please add `"x-ms-parameter-location": "method"`

---
In reply to: [166487297](https://github.com/Azure/azure-rest-api-specs/pull/2435#discussion_r166487297) [](ancestors = 166487297)

Reply:
After consult with team member, will leave this as Client